### DISCUSSION TITLE
"Bad URI syntax" error in Collabora container

### DISCUSSION THREAD

**Discussion by blendersamsonov** [nextcloud/all-in-one#917] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Deploy Nextcloud AIO container, install Collabora Office
2. Open any .doc or .odt file
3. See "Document loading failed" error. Office container log contains the following lines:
```
wsd-00001-00038 2022-07-11 10:13:58.220857 +0300 [ websrv_poll ] ERR  #52 Exception while processing incoming request: [GET /cool/https:/mycloud.mydomain.com/index.php/apps/richdocuments/wopi/files/898_ocjmu39drxnk%3Faccess_token=YDV31fAdVMcwmRIxldpmNlkbmRkzc23i&access_token_ttl=0&permission=edit/ws?WOPISrc=https%3A%2F%2Fmycloud.mydomain.com%2Findex.php%2Fapps%2Frichdocuments%2Fwopi%2Ffiles%2F898_ocjmu39drxnk&compat=/ws HTTP/1.1
...]: Bad URI syntax| wsd/COOLWSD.cpp:3539
```
Where the real address is substituted with `mycloud.mydomain.com`

The request should probably not include `https:/mycloud.mydomain.com/` and start as `/cool/index.php/apps/richdocuments/wopi/files/`

### Host OS 
Debian 11

#### Nextcloud AIO version 
1.4.2

#### Current channel 
latest

**Comment by szaimen** [comment, marked as answer]
I am sorry but I cannot reproduce this. It is likely something with your specific setup that you need to fix.

**Reply by binarynoise** [reply]
I'm having the same issue.
How do we find out what to fix?

**Reply by szaimen** [reply]
you probably need to fix your reverse proxy...

**Reply by binarynoise** [reply]
In general my nginx reverse proxy works, just not for Collabora Office.

How would the URL normally look like (that's probably the part that gets messed up on the way)

**Reply by binarynoise** [reply]
Looking around in the reverse proxy docs for another time, I finally found my mistake.
I forgot to append `$request_uri` to `proxy_pass`.

Interesting, that this mistake doesn't crash everything else

**Reply by leomeinel** [reply]
@binarynoise Hi, I am facing a very similar issue running CODE inside docker and behind a reverse proxy (actually 2 reverse proxies) that is proxied through cloudflare.
As far as I know, $request_uri is appended automatically to proxy_pass. Therefore I am wondering why you could solve it by appending it manually.
For me your solution didn't work. I also tried a lot of other stuff. The last time I worked on this was [half a year ago](https://forum.collaboraonline.com/t/nextcloud-office-cant-connect-to-code-behind-reverse-proxy-requesting-address-is-denied/1453/1).

When using more or less the [recommended reverse proxy config](https://sdk.collaboraonline.com/docs/installation/Proxy_settings.html#reverse-proxy-settings-in-nginx-config-ssl-termination) I am getting these errors and a very weird looking URI, maybe @szaimen may know what's wrong here:
```
wsd-00001-00028 2023-05-25 19:46:47.292178 +0000 [ websrv_poll ] WRN  convert-to: Requesting address is denied: <server's-public(non cloudflare)-ip>| wsd/COOLWSD.cpp:3704
wsd-00001-00028 2023-05-25 19:46:58.407373 +0000 [ websrv_poll ] ERR  #27: #27 Exception while processing incoming request: [GET /cool/https:/nc.meinel.link/index.php/apps/richdocuments/wopi/files/385660_ocuweivjs5xs%3Faccess_token=maWQJx2ZG6X7lIOkOdLxFQ3slqLs26Dv&access_token_ttl=1685080017000/ws?WOPISrc=https%3A%2F%2Fnc.meinel.link%2Findex.php%2Fapps%2Frichdocuments%2Fwopi%2Ffiles%2F385660_ocuweivjs5xs&compat=/ws HTTP/1.0...]: Bad URI syntax| wsd/COOLWSD.cpp:3952
wsd-00001-00028 2023-05-25 19:46:58.947525 +0000 [ websrv_poll ] ERR  #27: #27 Exception while processing incoming request: [GET /cool/https:/nc.meinel.link/index.php/apps/richdocuments/wopi/files/385660_ocuweivjs5xs%3Faccess_token=maWQJx2ZG6X7lIOkOdLxFQ3slqLs26Dv&access_token_ttl=1685080017000&permission=edit/ws?WOPISrc=https%3A%2F%2Fnc.meinel.link%2Findex.php%2Fapps%2Frichdocuments%2Fwopi%2Ffiles%2F385660_ocuweivjs5xs&compat=/ws HTTP/1.0...]: Bad URI syntax| wsd/COOLWSD.cpp:3952
wsd-00001-00001 2023-05-25 19:47:52.725467 +0000 [ coolwsd ] WRN  #29: Timed out while requesting [GET rating.collaboraonline.com/UpdateCheck/updatecheck2.html?product=Collabora%20Online%20Development%20Edition&version=22.05.14.3] after 30007ms| net/HttpRequest.hpp:1472
```

I am sorry for pinging if this isn't relevant to you btw. I'm just getting frustrated and didn't want to open another issue that is probably gonna be ignored or closed anyways.

Just as a sidenote: https://office.subdomain.tld/browser/dist/admin/admin.html is reachable and loads correctly. Therefore I can only assume that the reverse proxy is kind of working.

**Reply by binarynoise** [reply]
Using wireshark, I just looked into what the succeeding request looks like.

~~Comparing with the URL you sent, the difference is that my stuff after /cool is escaped and yours is not.
So in contrast to what we believed, the URL does belong there in its full form, but escaped correctly.
Your challenge would be to find out where the escaping goes missing.~~

Ah nevermind, the urls in the log are always unescaped.

What my browser requests: `wss://cloud.mycloud.de/cool/https%3A%2F%2Fcloud.mycloud.de%2Findex.php%2Fapps%2Frichdocuments%2Fwopi%2Ffiles%2F14427_ocnk4q1q8aky%3Faccess_token%3DWOoq4LNLarq8ZpTAHh1hWNQ7FpEmKqjB%26access_token_ttl%3D1685084421000/ws?WOPISrc=https%3A%2F%2Fcloud.mycloud.de%2Findex.php%2Fapps%2Frichdocuments%2Fwopi%2Ffiles%2F14427_ocnk4q1q8aky&compat=/ws`

My nginx passes this though 1:1 and responds with 101 Switching Protocols.
However, the next thing happening is nginx (not Firefox) calling the url that was encoded in the first request: `/index.php/apps/richdocuments/wopi/files/14427_ocnk4q1q8aky?access_token=WOoq4LNLarq8ZpTAHh1hWNQ7FpEmKqjC&access_token_ttl=1685084421000`




This is my current nginx config for my cloud:

```Nginx
    server {
        server_name cloud.mycloud.de;

        listen 443 ssl http2;
        listen [::]:443 ssl http2;

        ssl_certificate /etc/letsencrypt/live/cloud.mycloud.de/fullchain.pem; # managed by Certbot
        ssl_certificate_key /etc/letsencrypt/live/cloud.mycloud.de/privkey.pem; # managed by Certbot
        include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot

        location / {
            proxy_pass http://192.168.50.100:80$request_uri;
            proxy_http_version 1.1;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection $connection_upgrade;
            include snippets/reverse-proxy.conf;
            client_max_body_size 1G;
        }

        location /.well-known/carddav {
            return 301 $scheme://$host/remote.php/dav;
        }
        location /.well-known/caldav {
            return 301 $scheme://$host/remote.php/dav;
        }
        access_log off;
    }
```

**Reply by leomeinel** [reply]
Thank you very much for the answer and especially doing the work to give me a working URL example! You're awesome :)
I'll work on this in the following days and try to lyk if I found a way to solve my issues in case anyone else finds this!

**Reply by varioous** [reply]
Hi @leomeinel, today i had a similar issue with reverse proxy and collabora. For me the solution was to set the domain name 
`coolconfig set server_name office.example.com.`

Source: https://sdk.collaboraonline.com/docs/installation/Collabora_Online_Troubleshooting_Guide.html

Hope this might also help you.

**Reply by leomeinel** [reply]
@varioous Thanks for giving me another option. I didn't really have the time to work on this. But if I fix it, I'll post an update here!

**Comment by szaimen** [comment]
Moving to discussions
